dds/DCPS/NetworkConfigModifier.cpp

```diff
@@ -100,6 +100,79 @@ bool NetworkConfigModifier::close()
   return true;
 }
 
+void NetworkConfigModifier::update_interfaces()
+{
+  ifaddrs* p_ifa = 0;
+  ifaddrs* p_if = 0;
+
+  if (::getifaddrs(&p_ifa) != 0)
+    return;
+
+  std::map<std::string, ifaddrs*> net_names;
+
+  // Pull the address out of each INET interface.
+  for (p_if = p_ifa; p_if != 0; p_if = p_if->ifa_next) {
+    if (p_if->ifa_addr == 0)
+      continue;
+
+    // Check to see if it's up.
+    if ((p_if->ifa_flags & IFF_UP) != IFF_UP)
+      continue;
+
+    if (p_if->ifa_addr->sa_family == AF_INET) {
+      struct sockaddr_in* addr = reinterpret_cast<sockaddr_in*> (p_if->ifa_addr);
+      // Sometimes the kernel returns 0.0.0.0 as the interface
+      // address, skip those...
+      if (addr->sin_addr.s_addr != INADDR_ANY) {
+        net_names.insert(std::make_pair(p_if->ifa_name, p_if));
+      }
+    }
+# if defined (ACE_HAS_IPV6)
+    else if (p_if->ifa_addr->sa_family == AF_INET6) {
+      struct sockaddr_in6 *addr = reinterpret_cast<sockaddr_in6 *> (p_if->ifa_addr);
+      // Skip the ANY address
+      if (!IN6_IS_ADDR_UNSPECIFIED(&addr->sin6_addr)) {
+        net_names.insert(std::make_pair(p_if->ifa_name, p_if));
+      }
+    }
+# endif /* ACE_HAS_IPV6 */
+  }
+  // Remove interfaces that are no longer active
+  NetworkInterfaces nis = get_interfaces();
+  for (OpenDDS::DCPS::NetworkInterfaces::iterator iter = nis.begin(); iter < nis.end(); ++iter) {
+    if (net_names.find(iter->name()) == net_names.end()) {
+      remove_interface(iter->index());
+    }
+  }
+
+  // Add interfaces that are new
+  nis = get_interfaces();
+  for (std::map<std::string, ifaddrs*>::iterator iter = net_names.begin(); iter != net_names.end(); ++iter) {
```

Comment:
```suggestion
  for (Names::iterator iter = net_names.begin(); iter != net_names.end(); ++iter) {
```

Or whatever the type is.